src/Tickets/Blocks/Controller.php

```diff
@@ -100,14 +100,25 @@ public function register_for_blocks() {
 	 * Register the blocks after plugins are fully loaded.
 	 *
 	 * @since 5.3.0
+	 * @since TBD Correctly get post type when creating a new post or page.
 	 */
 	public function register_blocks() {
 		if ( is_admin() ) {
 			// In admin context, do not register the blocks if the post type is not ticketable.
-			$post      = get_post() ?: get_post( tribe_get_request_var( 'post' ) );
-			$post_type = get_post_type( $post ) ?: tribe_get_request_var( 'post_type' );
+			global $pagenow;
+			// Determine post type for new posts on 'post-new.php' or existing posts.
+			if ( 'post-new.php' === $pagenow ) {
+				// Attempt to get the post type from the request, defaulting to 'post'.
+				$post_type = tribe_get_request_var( 'post_type', 'post' );
+			} else {
+				// For existing posts, get the post object and determine its type.
+				$post      = get_post( tribe_get_request_var( 'post' ) );
+				$post_type = $post ? get_post_type( $post ) : tribe_get_request_var( 'post_type' );
+			}
```

Comment:
I would love to see this part of the code rewritten to avoid the global variable and the else.